### PR TITLE
added arg-parsing option for using Klett or Raman method

### PR DESCRIPTION
**Pull request by ulysses78**

### CODE REVIEW

**Review by ZPYin — COMMENTED** [review]
First of all, thanks for your efforts with implementing this feature.

I wonder whether this feature can be included without adding this new argurement `method`. Because in the current framework, I have parsed the retrieving method by going through the `*info.txt` file (configuration file output by Holger's Labview program). Please see the lines below:

https://github.com/PollyNET/polly2earlinetDB/blob/cc5af9b7c5e3c1bc5f684200fe2a94ea17fa68d5/src/polly2scc.py#L475-L482

By parsing the `*.info.txt` file, the retrieving method can be automatically determined. You can use `labviewInfo['retrieving_method']` to check whether it's **Raman** or **Klett**, instead of using the user input.

My question is whether it's possible to build this PR upon the previous framework. Because this would not change the calling interface and keep things easier.

**Comment by ulysses78** [comment]
I forgot to mention, that my implemented arg-parsing option (Klett/Raman) only works for picasso file-input. I didn't touched the labview-section. For the labview-section your suggestion could work.

**Review by ZPYin — APPROVED** [review]
Thanks for your clarifications. Let's merge it. 😄

**Comment by ZPYin** [comment]
The [Continuous Integration tool](https://en.wikipedia.org/wiki/Continuous_integration)(CI) shows errors for the coding style. For each PR, I have configured the coding style check and unit-testing. Only after passing such examinations, the PR can be accepted.

To my point of view, coding style check is very essential for code readability, especially for collaborations. Please have a read of the python coding style [PEP8](https://pypi.org/project/pycodestyle/). And revise your code according to the CI report (see how to open the CI report below).

image

Below are the coding style errors for this PR

``` bash
Run pycodestyle --max-line-length=80 --statistics src
src/polly2scc.py:103:81: E501 line too long (95 > 80 characters)
src/polly2scc.py:118:81: E501 line too long (84 > 80 characters)
src/polly2scc.py:1058:1: W293 blank line contains whitespace
src/polly2scc.py:1059:31: E225 missing whitespace around operator
src/polly2scc.py:1092:33: E225 missing whitespace around operator
src/polly2scc.py:1130:1: W293 blank line contains whitespace
src/polly2scc.py:1132:9: E303 too many blank lines (2)
src/polly2scc.py:1262:1: W293 blank line contains whitespace
src/polly2scc.py:1272:1: W293 blank line contains whitespace
src/polly2scc.py:1276:1: W293 blank line contains whitespace
src/polly2scc.py:1280:1: W293 blank line contains whitespace
src/polly2scc.py:1284:1: W293 blank line contains whitespace
src/polly2scc.py:1288:1: W293 blank line contains whitespace
src/polly2scc.py:1291:81: E501 line too long (81 > 80 characters)
src/polly2scc.py:1299:1: W293 blank line contains whitespace
src/polly2scc.py:1320:1: W293 blank line contains whitespace
src/polly2scc.py:1330:1: W293 blank line contains whitespace
src/polly2scc.py:1334:1: W293 blank line contains whitespace
src/polly2scc.py:1338:1: W293 blank line contains whitespace
src/polly2scc.py:1342:1: W293 blank line contains whitespace
src/polly2scc.py:1346:1: W293 blank line contains whitespace
src/polly2scc.py:1349:81: E501 line too long (81 > 80 characters)
src/polly2scc.py:1387:1: W293 blank line contains whitespace
src/polly2scc.py:1397:1: W293 blank line contains whitespace
src/polly2scc.py:1401:1: W293 blank line contains whitespace
src/polly2scc.py:1405:1: W293 blank line contains whitespace
src/polly2scc.py:1409:1: W293 blank line contains whitespace
src/polly2scc.py:1413:1: W293 blank line contains whitespace
src/polly2scc.py:1416:81: E501 line too long (81 > 80 characters)
src/polly2scc.py:1424:1: W293 blank line contains whitespace
src/polly2scc.py:1445:1: W293 blank line contains whitespace
src/polly2scc.py:1448:31: E127 continuation line over-indented for visual indent
src/polly2scc.py:1455:1: W293 blank line contains whitespace
src/polly2scc.py:1459:1: W293 blank line contains whitespace
src/polly2scc.py:1463:1: W293 blank line contains whitespace
src/polly2scc.py:1467:1: W293 blank line contains whitespace
src/polly2scc.py:1471:1: W293 blank line contains whitespace
src/polly2scc.py:1474:81: E501 line too long (81 > 80 characters)
src/polly2scc.py:1512:1: W293 blank line contains whitespace
src/polly2scc.py:1522:1: W293 blank line contains whitespace
src/polly2scc.py:1526:1: W293 blank line contains whitespace
src/polly2scc.py:1530:1: W293 blank line contains whitespace
src/polly2scc.py:1534:1: W293 blank line contains whitespace
src/polly2scc.py:1538:1: W293 blank line contains whitespace
src/polly2scc.py:1541:81: E501 line too long (83 > 80 characters)
src/polly2scc.py:1549:1: W293 blank line contains whitespace
src/polly2scc.py:1570:1: W293 blank line contains whitespace
src/polly2scc.py:1580:1: W293 blank line contains whitespace
src/polly2scc.py:1584:1: W293 blank line contains whitespace
src/polly2scc.py:1588:1: W293 blank line contains whitespace
src/polly2scc.py:1592:1: W293 blank line contains whitespace
src/polly2scc.py:1596:1: W293 blank line contains whitespace
src/polly2scc.py:1599:81: E501 line too long (85 > 80 characters)
src/polly2scc.py:1600:16: E114 indentation is not a multiple of 4 (comment)
src/polly2scc.py:1600:81: E501 line too long (99 > 80 characters)
src/polly2scc.py:1611:9: E303 too many blank lines (2)
src/polly2scc.py:1634:35: E225 missing whitespace around operator
src/polly2scc.py:1697:37: E225 missing whitespace around operator
src/polly2scc.py:1698:18: E111 indentation is not a multiple of 4
src/polly2scc.py:1698:18: E117 over-indented
src/polly2scc.py:1700:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1702:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1704:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1706:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1708:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1710:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1712:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1714:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1716:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1718:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1720:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1722:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1724:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1726:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1728:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1730:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1732:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1734:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1736:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1738:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1740:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1742:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1744:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1746:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1748:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1750:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1752:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1754:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1756:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1758:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1760:25: E131 continuation line unaligned for hanging indent
src/polly2scc.py:1761:17: E122 continuation line missing indentation or outdented
src/polly2scc.py:1874:35: E225 missing whitespace around operator
src/polly2scc.py:1937:37: E225 missing whitespace around operator
src/polly2scc.py:2114:35: E225 missing whitespace around operator
src/polly2scc.py:2168:13: E122 continuation line missing indentation or outdented
src/polly2scc.py:2169:37: E225 missing whitespace around operator
src/polly2scc.py:2238:13: E122 continuation line missing indentation or outdented
src/polly2scc.py:2568:81: E501 line too long (86 > 80 characters)
src/polly2scc.py:2672:81: E501 line too long (117 > 80 characters)
1       E111 indentation is not a multiple of 4
1       E114 indentation is not a multiple of 4 (comment)
1       E117 over-indented
3       E122 continuation line missing indentation or outdented
1       E127 continuation line over-indented for visual indent
31      E131 continuation line unaligned for hanging indent
8       E225 missing whitespace around operator
2       E303 too many blank lines (2)
11      E501 line too long (95 > 80 characters)
41      W293 blank line contains whitespace
```

> [pycodestyle](https://pypi.org/project/pycodestyle/) is a python package for code style check offline. And many modern editors also support linting, e.g., **vscode** or **sublime text**. 

I guess this might sound weired for a very tiny project. But trust me! This is one of the best practices that I have learned after suffering too much from reading others' code...

**Comment by ulysses78** [comment]
Ok. I am on it. Nice tool this pycodestyle python package. I work with that now to get nice python-code ;-)

**Comment by ulysses78** [comment]
I finished pycodestyle!

File: tests/test_polly_2_earlinet_convertor.py [hunk 1]
```diff
@@ -196,7 +196,7 @@ def test_polly2scc(self):
         print('---> Test on p2e_go')
 
         polly2scc(
-            'PollyXT_TROPOS', 'leipzig', 'picasso', 1,
+            'PollyXT_TROPOS', 'leipzig', 'picasso', 1, 'raman',
```
Comment (ZPYin) [inline tests/test_polly_2_earlinet_convertor.py:199]: I changed a little bit of the unit-test to suit to your implementation.

**Comment by ZPYin** [comment]
Hi Andi,

Thanks for your PR and fast response. It's ready to be merged now.

Best regards,
Zhenping

**Pull request closed without merging**